src/document.c

```diff
@@ -472,42 +495,59 @@ FIELD_BULK_INDEXER(numericIndexer) {
 }
 
 FIELD_PREPROCESSOR(geoPreprocessor) {
-  // buffer of 32 bytes should be sufficient. lon,lat 
   size_t len;
-  char buf[32];
-  const char *str = RedisModule_StringPtrLen(field->text, &len);
-  if (len > 31) {
-    return REDISMODULE_ERR;
-  }
-
-  // copy to buffer since we modify the string
-  memcpy(buf, str, MIN(32, len));
-  buf[len] = '\0';
+  const char *str = NULL;
+  double lat = 0, lon = 0;
+
+  switch (field->unionType) {
+    case FLD_VAR_T_GEO:
+      lon = field->lon;
+      lat = field->lat;
+      break;
+    case FLD_VAR_T_CSTR:
+    case FLD_VAR_T_RMS:
+      str = DocumentField_GetValueCStr(field, &len);
+      if (len > 31) {
+        return REDISMODULE_ERR;
+      }
 
-  char *pos = strpbrk(buf, " ,");
-  if (!pos) {
-    QueryError_SetCode(status, QUERY_EGEOFORMAT);
-    return -1;
-  }
-  *pos = '\0';
-  pos++;
+      // buffer of 32 bytes should be sufficient. lon,lat
+      char buf[32];
+      // copy to buffer since we modify the string
+      memcpy(buf, str, len);
+      buf[len] = '\0';
 
-  char *end1 = NULL, *end2 = NULL;
-  double lon = strtod(buf, &end1);
-  double lat = strtod(pos, &end2);
-  if (*end1 || *end2) {
-    return REDISMODULE_ERR;
+      char *pos = strpbrk(buf, " ,");
```

Comment:
Why not using parseGeo here? Also you have an assumption that there will always be a ' ' (space) before the ','?

Reply:
1. Good idea :+1: 
2. No. `strpbrk` takes the characters which are accepted. Either ` `  or `,`.